app/models/input_template_renderer.rb

```diff
@@ -40,7 +49,49 @@ def input(name)
     end
   end
 
+  def render_template(template_name, input_values = {}, options = {})
+    options.assert_valid_keys(:with_foreign_input_set)
+    with_foreign_input_set = options.fetch(:with_foreign_input_set, true)
+    template = JobTemplate.authorized(:view_job_templates).where(:name => template_name).first
+    unless template
+      self.error_message = _('included template \'%s\' not found') % template_name
+      raise error_message
+    end
+    if with_foreign_input_set
+      input_values = foreign_input_set_values(template, input_values)
+    end
+    included_renderer = self.class.new(template, host, invocation, input_values, @preview, @templates_stack)
+    out = included_renderer.render
+    if included_renderer.error_message
+      self.error_message = included_renderer.error_message
+      raise error_message
+    else
+      out
+    end
+  end
+
+  def foreign_input_set_values(target_template, overrides = {})
+    input_set = @template.foreign_input_sets.where(:target_template_id => target_template).first
+    return {} if input_set.nil?
+    if input_set.include_all?
+      inputs_to_generate = input_set.target_template.template_inputs.pluck(:name)
+    else
+      inputs_to_generate = input_set.include.to_s.split(',').map(&:strip)
+    end
+    inputs_to_generate -= overrides.keys + input_set.exclude.to_s.split(',').map(&:strip)
+
+    included_renderer = self.class.new(input_set.target_template, host, invocation, nil, @preview, @templates_stack)
+    input_values = inputs_to_generate.inject({}) do |hash, input_name|
```

Comment:
this might be worth of extracting to a input_values method of the renderer itself